### PR TITLE
Feature lwdev 6978 apply consistent number formatting throughout the app

### PR DESCRIPTION
**Pull request by ArtLait**

### CODE REVIEW

File: src/utils/localFormatted/localFormatted.spec.ts [hunk 1]
```diff
@@ -1,27 +1,19 @@
-import {formattedDate, formattedNumber} from './localFormatted';
+import {formattedNumber} from './localFormatted';
 
+const locale = 'en-US';
```
Comment (optimusway) [inline src/utils/localFormatted/localFormatted.spec.ts]: this is a dirty hack and will fail on CI running in non-US env

File: src/utils/localFormatted/localFormatted.spec.ts [hunk 2]
```diff
@@ -1,27 +1,19 @@
...
-      expect(formattedNumber('50000.05', 4)).toBe(
-        results[locale].expectNumberWithAccuracy
-      );
-    });
-
-    it('should return string date with locale', () => {
-      expect(formattedDate(date)).toBe(results[locale].expectDate);
-    });
-  }
+  it('should return string number with locale', () => {
```
Comment (optimusway) [inline src/utils/localFormatted/localFormatted.spec.ts]: plz add more tests

i see at least 4-5 cases in the task multiplied by edge values for each case, like NaN, Infinity, 0, etc

File: src/utils/localFormatted/localFormatted.ts [hunk 3]
```diff
@@ -1,47 +1,35 @@
-export const locale =
-  (navigator.languages && navigator.languages[0]) ||
-  navigator.language ||
-  'en-US';
+export function formattedNumber(
```
Comment (optimusway) [inline src/utils/localFormatted/localFormatted.ts]: `formatWithAccuracy` would be better + you need to extend the implementation to make it possible to pad fraction part with zeroes, as discussed today

this is the case for the order book, where we need _1.12_ --> _1.12000_, given accuracy is 5

File: src/utils/localFormatted/localFormatted.ts [hunk 4]
```diff
@@ -1,47 +1,35 @@
...
+  value: any,
```
Comment (optimusway) [inline src/utils/localFormatted/localFormatted.ts]: nor `any`, nor `string`, no implicits, we should be strict here
only a `number` type is possible

see original `toLocaleString` impl for the reference

File: src/utils/localFormatted/localFormatted.ts [hunk 5]
```diff
@@ -1,47 +1,35 @@
...
+  accuracy?: number,
+  options?: any
```
Comment (optimusway) [inline src/utils/localFormatted/localFormatted.ts]: `options?: object`

File: src/utils/localFormatted/localFormatted.ts (hunk 3 above)
Comment (optimusway) [inline src/utils/localFormatted/localFormatted.ts]: plz use fat-arrow function instead of `function`

File: src/utils/localFormatted/localFormatted.ts [hunk 6]
```diff
@@ -1,47 +1,35 @@
...
+): string {
+  const indexOfPoint = (value + '').search(/\./);
```
Comment (optimusway) [inline src/utils/localFormatted/localFormatted.ts]: no need for regex here

File: src/utils/localFormatted/localFormatted.ts [hunk 7]
```diff
@@ -1,47 +1,35 @@
...
-  const options = {
-    hour: 'numeric',
-    minute: 'numeric',
-    second: 'numeric'
-  };
-  return new Intl.DateTimeFormat(locale, options).format(date);
-}
+  let result = value.toLocaleString(undefined, options);
+  const indexOfZero = result.search(/0+$/);
+  result = indexOfZero !== -1 ? result.slice(0, indexOfZero + 1) : result + '0';
```
Comment (optimusway) [inline src/utils/localFormatted/localFormatted.ts]: as i see it will fail in the case `formatWithAccuracy(1.12345, 5)`
it will produce _1.123450_ but _1.12345_ is expected

am i wrong?

File: src/utils/localFormatted/localFormatted.ts [hunk 8]
```diff
@@ -1,47 +1,35 @@
-export const locale =
-  (navigator.languages && navigator.languages[0]) ||
-  navigator.language ||
-  'en-US';
+export function formattedNumber(
+  value: any,
+  accuracy?: number,
```
Comment (optimusway) [inline src/utils/localFormatted/localFormatted.ts]: accuracy is mandatory arg

File: src/utils/localFormatted/localFormatted.spec.ts (hunk 1 above)
Comment (optimusway) [reply src/utils/localFormatted/localFormatted.spec.ts]: what you need to do is to `export default` curried function with the `locale` setted to `undefined` along with the named export

then the original function will be pure and you can easily test it, with the named import, and the app itself will use default export which is curried

File: src/utils/localFormatted/localFormatted.ts (hunk 7 above)
Comment (ArtLait) [reply src/utils/localFormatted/localFormatted.ts]: e.g. 1.23 --> 1.230
